lib/bloc/pos_catalog/sqlite/repository.dart

```diff
@@ -198,4 +198,36 @@ class SqliteRepository implements Repository {
       ),
     );
   }
+
+  @override
+  Future<Set<Sale>> fetchSalesByTimeRange(DateTime start, DateTime end) async {
+    final db = await getDB();
+    final sales = await db.query(
+      "sale",
+      where: "date >= ? AND date <= ? ",
+      whereArgs: [
+        start.millisecondsSinceEpoch,
+        end.millisecondsSinceEpoch,
+      ],
+    );
+    final result = Set<Sale>();
+    for (final sale in sales){
+      final baseSale = Sale.fromMap(sale);
+
+      final saleLines = await _fetchDBItems(
```

Comment:
So here we are going over all sales and fetching their sale lines one by one which is not an effective use of sqlite.
Instead let's see if we can let the sql lite engine do the work by querying sale lines table directly for what we need as the final result, aggregating the total sats and fiat values.
For example something like this:

`Select sum( (quantity * price_per_item)) as fiat_value, sum( (quantity * price_per_item * sat_conversion_rate) ) as sat_value, count(sale_id) as total_sales from sale_lines left join sales on sale_lines.sale_id = sales.id where (sales.date < ? and sales.date > ?)`

We can create a model for this report item and get it directly in the UI.

Reply:
I understood now, I had to make some changes:

```sql
SELECT
  sum((quantity * price_per_item)) as fiat_value,
  sum((quantity * price_per_item * sat_conversion_rate)) as sat_value,
  count(sale_id) as total_sales
FROM sale_line
LEFT JOIN sale ON sale_line.sale_id = sale.id
WHERE (sale.date >= ? and sale.date <= ?);
```

But there is a problem, It is not checking the fiat currency type. i.e. it sums all together sales made in USD, MXN, EUR, etc…

The Sale model has the correct logic:

```dart
  Map<String, double> get totalAmountInFiat {
    Map<String, double> totals = {};
    saleLines.forEach((saleLine) {
      final previous = totals[saleLine.currency] ?? 0.0;
      totals[saleLine.currency] = previous + saleLine.totalFiat;
    });
    return totals;
  }

  double get totalAmountInSats {
    var total = 0.0;
    saleLines.forEach((saleLine) {
      total += saleLine.totalSats;
    });
    return total;
  }
```
I'm not sure if it is possible to replicate this logic in pure SQL but even if it is possible I don't know if it is a good idea.

Reply:
I think adding group by currency to the SQL query will do the job.
IMO imementing this in SQL is the right thing. It uses SQLite engine, code more compact and far better performance.

Reply:
OK, I'll study this SQL group thing, I never did that.

Reply:
If you group by fiat currency, remember we need to display the sats amount of all the groups (including an empty group) and the fiat amount only if the group by returns one group. I haven't done sql in a while though...

Reply:
I did add the GROUP BY but I had to add a complementary code too, I'm not literate on SQL, if you have some idea on how to improve it please share it.

https://github.com/breez/breezmobile/pull/799/files#diff-9ef5e59fe1bc906213338b91d9ddd81cd09fe7ef8d5233f6004cc19173f08a4dR203